### PR TITLE
Refactor: Carousel refactor

### PR DESCRIPTION
**Pull request by JUNIOR140889**
This PR includes:

 * Carousel refactor
 
Evidence:

https://github.com/JUNIOR140889/Black-Market-RN/assets/42948405/74af2dca-c4bd-4a72-8841-82a8c080e489

### CODE REVIEW

File: src/modules/home/components/carousel-item.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        <View className="items-center">
+          <Image source={item.pictures[0]} className="m-2 h-32 w-32 rounded-md object-cover" />
+        </View>
+        <View className="h-px bg-black shadow-lg shadow-gray-400 " />
+        <View className="border-t-1 flex-row items-center justify-between border-t border-black px-3 pt-3 opacity-100">
+          <Text variant="h6">{item.unitPrice}</Text>
+          <View
+            className={`${item.state === 'totaly_new' ? 'bg-tags-new' : 'bg-tags-restored'} flex items-center justify-center rounded-md p-1 px-2`}>
+            <Text className="text-white" variant="body1">
+              {`${item.state === 'totaly_new' ? 'New' : 'Restored'}`}
```
Comment (fernandatoledo) [inline src/modules/home/components/carousel-item.tsx]: New and restored should be localized

**Review by fernandatoledo — APPROVED** [review]

File: src/modules/home/components/carousel-item.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import type { StackNavigationProp } from '@react-navigation/stack';
+import React from 'react';
+
+import type { Product } from '../../../api/types';
+import TabFavoriteIcon from '../../../ui/assets/svgs/tab-favorite-icon';
+import { Image } from '../../../ui/core/nativewind/image';
+import { Text } from '../../../ui/core/text';
+import { TouchableOpacity, View } from '../../../ui/core/view';
+
+type RootStackParamList = {
```
Comment (ramirogavagnin) [inline src/modules/home/components/carousel-item.tsx]: I think this type belongs to the `navigation` folder

File: src/modules/home/components/carousel.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import { FlatList as RNFlatList } from 'react-native';
+
+import type { Product } from '../../../api/types';
+import { common } from '../../../translations/en.json';
+import { Text } from '../../../ui/core/text';
+import { View } from '../../../ui/core/view';
+import { CarouselItem } from './carousel-item';
+const FlatList = styled(RNFlatList);
+
+type Props = {
```
Comment (ramirogavagnin) [inline src/modules/home/components/carousel.tsx]: just a suggestion `HorizontalCarouselProps`

File: src/modules/home/components/carousel.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  lenght: number;
+};
+
+export const HorizontalCarousel = ({ data }: Props) => {
+  return (
+    <>
+      <FlatList
+        className="h-[300]"
+        data={data}
+        contentContainerStyle={{ alignItems: 'center', gap: 16, paddingHorizontal: 16 }}
```
Comment (ramirogavagnin) [inline src/modules/home/components/carousel.tsx]: in line styles ;)

**Review by ramirogavagnin — APPROVED** [review]
great job!!

**Review by juanchoperezj — APPROVED** [review]

File: src/modules/home/components/carousel-item.tsx (hunk 1 above)
Comment (JUNIOR140889) [reply src/modules/home/components/carousel-item.tsx]: Done.

File: src/modules/home/components/carousel-item.tsx (hunk 2 above)
Comment (JUNIOR140889) [reply src/modules/home/components/carousel-item.tsx]: Done.

File: src/modules/home/components/carousel.tsx (hunk 3 above)
Comment (JUNIOR140889) [reply src/modules/home/components/carousel.tsx]: Done.

File: src/modules/home/components/carousel.tsx (hunk 4 above)
Comment (JUNIOR140889) [reply src/modules/home/components/carousel.tsx]: I analyze the option of using Tailwind React Native Classnames library to style in the contentContainerStyle.

**Pull request merged**